scripts/metadataProvider/createPseudoModulesInfo.ts

```diff
@@ -0,0 +1,224 @@
+		if (enumEntry.experimentalInfo.text) {
+			experimental += ` - ${enumEntry.experimentalInfo.text}`;
+		}
+		jsDocBuilder.push(experimental);
+	}
+
+	if (enumEntry.deprecatedInfo) {
+		let deprecated = `${indent} * @deprecated`;
+		if (enumEntry.deprecatedInfo.since) {
+			deprecated += ` (since ${enumEntry.deprecatedInfo.since})`;
+		}
+		if (enumEntry.deprecatedInfo.text) {
+			deprecated += ` - ${enumEntry.deprecatedInfo.text}`;
+		}
+		jsDocBuilder.push(deprecated);
+	}
+
+	if (enumEntry.visibility) {
+		jsDocBuilder.push(`${indent} * @${enumEntry.visibility}`);
+	}
+
+	if (enumEntry.since) {
+		jsDocBuilder.push(`${indent} * @since ${enumEntry.since}`);
+	}
+	jsDocBuilder.push(`${indent}*/`);
+
+	return jsDocBuilder.join("\n");
+}
+
+async function addOverrides(enums: Record<string, {enum: UI5Enum; export: string}[]>) {
+	const indexFilesImports: string[] = [];
+
+	for (const libName of Object.keys(enums)) {
+		const enumEntries = enums[libName];
+		const stringBuilder: string[] = [];
+
+		enumEntries.forEach(({enum: enumEntry, export: exportName}) => {
+			if (enumEntry.kind !== "UI5Enum") {
+				return;
+			}
+
+			const exportNameChunks = exportName.split(".");
+			const name = exportNameChunks[0]; // Always import the first chunk and then export the whole thing
+
+			stringBuilder.push(`declare module "${libName.replaceAll(".", "/")}/${exportName.replaceAll(".", "/")}" {`);
+
+			stringBuilder.push(`\timport {${name}} from "${libName.replaceAll(".", "/")}/library";`);
+			stringBuilder.push("");
+			stringBuilder.push(buildJSDoc(enumEntry, "\t"));
+			stringBuilder.push(`\texport default ${exportName};`);
+
+			stringBuilder.push(`}`);
+			stringBuilder.push("");
+
+			return stringBuilder.join("\n");
+		});
+
+		indexFilesImports.push(`import "./${libName}";`);
+		await writeFile(
+			new URL(`../../resources/overrides/library/${libName}.d.ts`, import.meta.url),
+			stringBuilder.join("\n")
+		);
+	}
+
+	await writeFile(
+		new URL(`../../resources/overrides/library/index.d.ts`, import.meta.url),
+		indexFilesImports.join("\n") + "\n"
+	);
+}
+
+async function cleanup() {
+	const apiJsonList = await readdir(RAW_API_JSON_FILES_FOLDER);
+
+	await Promise.all(apiJsonList.map((library) => unlink(path.resolve(RAW_API_JSON_FILES_FOLDER, library))));
+}
+
+async function main(url: string, sapui5Version: string) {
+	await fetchAndExtractAPIJsons(url);
+
+	await transformFiles(sapui5Version);
+
+	await cleanup();
+}
+
+try {
+	const url = process.argv[2];
+	let sapui5Version: string | null | undefined = process.argv[3];
+
+	if (!url) {
+		throw new Error("Second argument \"url\" is missing");
+	}
+
+	if (!sapui5Version) {
+		// Try to extract version from url
+		const versionMatch = url.match(/\/\d{1}\.\d{1,3}\.\d{1,3}\//gi);
+		sapui5Version = versionMatch?.[0].replaceAll("/", "");
+	}
+	if (!sapui5Version) {
+		throw new Error("\"sapui5Version\" cannot be determined. Provide it as a second argument");
+	}
+
+	await main(url, sapui5Version);
```

Comment:
We now have two scripts that use the same data as input. Why about refactoring them into one script? Can also be done in a separate PR.

Reply:
Yes, it uses the same URL.

Reply:
Sorry, not the URL but the extracted zip which has to be downloaded from the same URL. IMO you can combine them and download the zip to be used for both use cases.

Reply:
Thanks!
I have moved the common functionality into an external, helper module